ipatests/test_integration/test_cert.py

```diff
@@ -90,3 +92,48 @@ def test_ipa_getcert_san_aci(self):
         assert dnsnames == [self.clients[0].hostname]
         ipaddrs = ext.value.get_values_for_type(x509.IPAddress)
         assert ipaddrs == [ipaddress.ip_address(self.clients[0].ip)]
+
+
+class TestCertmongerInterruption(IntegrationTest):
+    num_replicas = 1
+
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(cls.master)
+        tasks.install_replica(cls.master, cls.replicas[0])
+
+    def test_certmomger_tracks_renewed_certs_during_interruptions(self):
+        """Test to check if Certmonger tracks certs in between
+        reboots/interruptions and while in "CA_WORKING" state
+
+        A non-renewal master CA might submit a renewal request before
+        the renewal master actually updating the certs. This is expected.
+
+        But, if certmonger on this node is stopped/interrupted
+        for any reason while the resubmit request is in "CA_WORKING" state
+        it will not be able to track changes to this cert anymore.
+
+        Pagure Issue: https://pagure.io/freeipa/issue/8164
+        """
+        cmd = ['getcert', 'resubmit', '-f',
+               paths.RA_AGENT_PEM]
+        result = self.replicas[0].run_command(cmd)
+        time.sleep(15)
```

Comment:
I suppose this is to wait for the request to be in CA_WORKING. Rather than sleeping for 15s, it might be more reliable to poll until a timeout is reached.

Reply:
@rcritten 's patch addresses this